tasks/run_cd4pe_job.rb

```diff
@@ -0,0 +1,71 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+
+require 'json'
+require 'rubygems/package'
+require 'open3'
+
+@params = JSON.parse(STDIN.read)
+require_relative File.join(@params['_installdir'], 'cd4pe_deployments', 'lib', 'puppet_x', 'puppetlabs', 'cd4pe_client')
+require_relative File.join(@params['_installdir'], 'cd4pe_deployments', 'lib', 'puppet_x', 'puppetlabs', 'gzip_helper')
+require_relative File.join(@params['_installdir'], 'cd4pe_deployments', 'lib', 'puppet_x', 'puppetlabs', 'cd4pe_job_helper')
+
+@working_dir = File.join(@params['_installdir'], 'cd4pe_job_working_dir')
+
+def docker_image
+  @params['docker_image']
+end
+
+def docker_run_args
+  @params["docker_run_args"]
+end
+
+def job_instance_id
+  @params["job_instance_id"]
+end
+
+def get_combined_exit_code(output)
+  job = output[:job]
+  after_job_success = output[:after_job_success]
+  after_job_failure = output[:after_job_failure]
+
+  exit_code_sum = job[:exit_code]
+  if (!after_job_success.nil?)
+    exit_code_sum = exit_code_sum + after_job_success[:exit_code]
+  end
+
+  if (!after_job_failure.nil?)
+    exit_code_sum = exit_code_sum + after_job_failure[:exit_code]
+  end
+
+  exit_code_sum == 0 ? 0 : 1
```

Comment:
But there are a ton of valid exit codes, not just 1 or 0!

Just pass the actual exit code around rather than force it to be 0 or 1, because the exit codes are meaningful as they represent different kinds of failures, where `1` is the most general "it failed".

Reply:
I agree! Copy pasta from Jonathan's question above:

for our purposes, we still end up exiting 1 regardless of the actual exit code. This is because we are actually running multiple scripts with each job, and this exit code represents the overall code we want Bolt to act upon. We do, however, return the actual exit code from each script in the json output that we print for Bolt such that the caller can still use that info if they want to (see the log output above for an example of how we capture multiple error codes).

Fun fact: Bolt prefers we print a single JSON object to STDOUT when running a bolt task in order for it to capture it properly: https://puppet.com/docs/bolt/latest/writing_tasks.html#returning-structured-output